internal/dag/builder.go

```diff
@@ -197,15 +197,13 @@ func (b *builder) addTCPService(svc *v1.Service, port *v1.ServicePort, strategy
 	return s
 }
 
+// lookupSecret returns a kuberntes secret if found or nil if not
 func (b *builder) lookupSecret(m meta, validate func(*v1.Secret) bool) *Secret {
 	if s, ok := b.secrets[m]; ok {
 		return s
 	}
 	sec, ok := b.source.secrets[m]
-	if !ok {
-		return nil
-	}
-	if !validate(sec) {
+	if !ok || !validate(sec) {
```

Comment:
I'm not sure this is an improvement in readability, i had to squint real hard to convince myself that if ok is true then the logic falls through to validate(sec)

Given that both validate functions, validSecret and validCA will work just fine, ie not panic, but return false if passed a nil. You could rewrite this as

```go
sec := b.source.secrets[m]
if !validate(sec) {
       return nil
}
```

Reply:
opps, my bad, this wasn't correct. if sec == nil then both validate functions will panic. My recommendation is to roll back the change to this function. thanks for the comment,

Reply:
I've revered this back to the way it was